avail-subxt/examples/max_block_submit.rs

```diff
@@ -0,0 +1,40 @@
+use avail_subxt::avail::runtime_types::frame_support::storage::bounded_vec::BoundedVec;
+use avail_subxt::*;
+use sp_keyring::AccountKeyring;
+use std::time::Instant;
+use subxt::ext::sp_core::H256;
+use subxt::{tx::PairSigner, OnlineClient};
+
+/// This example gets all the headers from testnet. It requests them in concurrently in batches of BATCH_NUM.
+/// Fetching headers one by one is too slow for a large number of blocks.
+
+#[async_std::main]
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    let signer = PairSigner::new(AccountKeyring::Alice.pair());
+
+    let api = OnlineClient::<AvailConfig>::new().await?;
+    let size: usize = 2 * 1024 * 1024;
+    let max_size: usize = 16 * 1024;
+    let num_chunks = size / max_size;
+    let mut hashes = Vec::<H256>::new();
+    let extrinsic_params = AvailExtrinsicParams::new_with_app_id(1);
+    let start = Instant::now();
+    for i in 1..=num_chunks {
+        let data_transfer =
+            avail::tx()
+                .data_availability()
+                .submit_data(BoundedVec(vec![(i & 255) as u8; max_size]));
+        let h = api
+            .tx()
+            .sign_and_submit(&data_transfer, &signer, extrinsic_params.clone())
+            .await
+            .unwrap();
+        println!("hash #{i}: {h}");
+        hashes.push(h);
+    }
+    let end = start.elapsed();
+
+    println!("Done in {end:?}!");
```

Comment:
Can we verify that block was filled up? May not be in this script but in a separate script to extract block size?

Reply:
Yes, it's not exactly for this script, but I've started doing this in the tests, as it requires a little more effort.